src/enclave/rpc_context.h

```diff
@@ -121,35 +121,13 @@ namespace enclave
     //
     // Only set in the case of a forwarded RPC
     //
-    struct Forwarded
-    {
-      // Initialised when forwarded context is created
-      const size_t client_session_id;
-      const ccf::CallerId caller_id;
-
-      Forwarded(size_t client_session_id_, ccf::CallerId caller_id_) :
-        client_session_id(client_session_id_),
-        caller_id(caller_id_)
-      {}
-    };
-    std::optional<Forwarded> original_caller = std::nullopt;
-
-    // Constructor used for non-forwarded RPC
+    bool is_forwarded = false;
```

Comment:
So I removed `caller_id` as mentioned in the description, and instead the cert is always present. And then `original_caller` was an optional struct containing only a `client_session_id`, but when this is filled the 'local' `client_session_id` above was invalid? So instead, we just reuse this same property for local or forwarded transactions. On the forwardee we do nothing with it other than returning it to the caller.